Core/Core/Conversations/ConversationListViewController.swift

```diff
@@ -0,0 +1,214 @@
+        } else if !conversations.pending {
+            emptyView.isHidden = true
+        }
+    }
+
+    func enrollmentsDidUpdate() {
+        if enrollments.pending == false && enrollments.requested {
+            filterButton.isEnabled = true
+        }
+    }
+
+    @IBAction func createNewConversation() {
+        env.router.show(ConversationCoursesActionSheet.create(delegate: self), from: self, options: .modal())
+    }
+
+    public func courseSelected(course: Course, user: User) {
+        let compose = ComposeViewController.create(
+            context: ContextModel(.course, id: course.id),
+            observeeID: user.id,
+            subject: course.name,
+            hiddenMessage: String.localizedStringWithFormat(
+                NSLocalizedString("Regarding: %@", bundle: .core, comment: ""),
+                user.name
+            )
+        )
+        env.router.show(compose, from: self, options: .modal(embedInNav: true))
+    }
+
+    @IBAction func actionFilterButtonPushed(_ sender: UIButton) {
+        if selectedCourse != nil {
+            updateSelectedCourse(nil)
+        } else {
+            let alert = UIAlertController(title: nil, message: NSLocalizedString("Filter by:", bundle: .core, comment: ""), preferredStyle: .actionSheet)
+            for e in enrollments {
+                guard let course = e.course else { continue }
+                alert.addAction(AlertAction(course.name, style: .default) { [weak self] _ in
+                    self?.updateSelectedCourse(course)
+                })
+            }
+            alert.addAction(AlertAction(NSLocalizedString("Cancel", bundle: .core, comment: ""), style: .cancel))
+            alert.popoverPresentationController?.sourceView = sender
+            alert.popoverPresentationController?.sourceRect = sender.bounds
+            env.router.show(alert, from: self, options: .modal())
+        }
+    }
+
+    func updateSelectedCourse(_ course: Course?) {
+        selectedCourse = course
+        let buttonTitle = selectedCourse == nil ? NSLocalizedString("Filter", comment: "") : NSLocalizedString("Clear Filter", comment: "")
+        filterButton.setTitle(buttonTitle, for: .normal)
+        refreshConversations(force: true)
+    }
+}
+
+extension ConversationListViewController: UITableViewDataSource, UITableViewDelegate {
+    public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return conversations.count
+    }
+
+    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell: ConversationListCell = tableView.dequeue(for: indexPath)
+        if let conversation = conversations[indexPath] {
+            cell.update(conversation)
+        }
+        return cell
+    }
+
+    public func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        if scrollView.isBottomReached() {
+            conversations.getNextPage()
+        }
```

Comment:
I think we've switched from this pattern to the LoadingCell pattern since that works more reliably.